### PR TITLE
[Docs]Fix docs about custom runtime and custom model

### PR DESCRIPTION
**Pull request by VVsssssk**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Fix docs about custom runtime and custom model, refer to https://github.com/open-mmlab/mmdetection/pull/8518
Add comments in schedule configs.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/advanced_guides/customize_runtime.md [hunk 1]
```diff
@@ -1,17 +1,42 @@
-# Customize Runtime Settings
+# Tutorial 5: Customize Runtime Settings
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:1]: Do we still need "Tutorial 5"?

**Review by Tai-Wang — COMMENTED** [review]
Please also adjust the correponding parts in the Chinese doc

**Comment by VVsssssk** [comment]
> Please also adjust the correponding parts in the Chinese doc


Do we need to modify chinese docs? I think it stays to the community to do.

File: docs/en/advanced_guides/customize_runtime.md [hunk 2]
```diff
@@ -20,240 +45,268 @@ To modify the learning rate of the model, the users only need to modify the `lr`
...
-
 ## Customize hooks
 
 ### Customize self-implemented hooks
 
 #### 1. Implement a new hook
 
-There are some occasions when the users might need to implement a new hook. MMDetection supports customized hooks in training (#3395) since v2.3.0. Thus the users could implement a hook directly in mmdet or their mmdet-based codebases and use the hook by only modifying the config in training.
-Before v2.3.0, the users need to modify the code to get the hook registered before training starts.
+MMEngine provides many useful [hooks](https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/hook.md), but there are some occasions when the users might need to implement a new hook. MMDetection3D supports customized hooks in training in v1.0 . Thus the users could implement a hook directly in mmdet3d or their mmdet3d-based codebases and use the hook by only modifying the config in training.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:256]: MMDetection3D supports customized hooks in training in v1.0 -> MMDetection3D supports customized hooks in training based on MMEngine after v1.1.0rc0

File: docs/en/advanced_guides/customize_runtime.md [hunk 3]
```diff
@@ -2,16 +2,41 @@
 
 ## Customize optimization settings
 
-### Customize optimizer supported by PyTorch
+Optimization related configuration is now all managed by `optim_wrapper` which usually has three fields: `optimizer`, `paramwise_cfg`, `clip_grad`, refer to [OptimWrapper](https://mmengine.readthedocs.io/en/latest/tutorials/optim_wrapper.md) for more detail. See the example below, where `Adamw` is used as an `optimizer`, the learning rate of the backbone is reduced by a factor of 10, and gradient clipping is added.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:5]: `clip_grad`. Please refer to xxxx

File: docs/en/advanced_guides/customize_runtime.md [hunk 4]
```diff
@@ -20,240 +45,268 @@ To modify the learning rate of the model, the users only need to modify the `lr`
 A customized optimizer could be defined as following.
 
 Assume you want to add a optimizer named `MyOptimizer`, which has arguments `a`, `b`, and `c`.
-You need to create a new directory named `mmdet3d/core/optimizer`.
-And then implement the new optimizer in a file, e.g., in `mmdet3d/core/optimizer/my_optimizer.py`:
+You need to create a new directory named `mmdet3d/engine/optimizers`. And then implement the new optimizer in a file, e.g., in `mmdet3d/engine/optimizers/my_optimizer.py`:
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:48]: named `mmdet3d/engine/optimizers`, and then implement

File: docs/en/advanced_guides/customize_runtime.md [hunk 5]
```diff
@@ -20,240 +45,268 @@ To modify the learning rate of the model, the users only need to modify the `lr`
...
 ```
 
-The module `mmdet3d.core.optimizer.my_optimizer` will be imported at the beginning of the program and the class `MyOptimizer` is then automatically registered.
+The module `mmdet3d.engine.optimizers.my_optimizer` will be imported at the beginning of the program and the class `MyOptimizer` is then automatically registered.
 Note that only the package containing the class `MyOptimizer` should be imported.
-`mmdet3d.core.optimizer.my_optimizer.MyOptimizer` **cannot** be imported directly.
+`mmdet3d.engine.optimizers.my_optimizer.MyOptimizer` **cannot** be imported directly.
 
-Actually users can use a totally different file directory structure in this importing method, as long as the module root can be located in `PYTHONPATH`.
+Actually users can use a totally different file directory structure using this importing method, as long as the module root can be located in `PYTHONPATH`.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:84]: using this importing method -> with this importing method

Comment (Tai-Wang) [reply docs/en/advanced_guides/customize_runtime.md:84]: root can be located -> root is located

File: docs/en/advanced_guides/customize_runtime.md [hunk 6]
```diff
@@ -20,240 +45,268 @@ To modify the learning rate of the model, the users only need to modify the `lr`
...
+          convert_to_iter_based=True)
+  ]
   ```
 
 ## Customize training schedules
 
-By default we use step learning rate with 1x schedule, this calls [`StepLRHook`](https://github.com/open-mmlab/mmcv/blob/v1.3.7/mmcv/runner/hooks/lr_updater.py#L167) in MMCV.
-We support many other learning rate schedule [here](https://github.com/open-mmlab/mmcv/blob/v1.3.7/mmcv/runner/hooks/lr_updater.py), such as `CosineAnnealing` and `Poly` schedule. Here are some examples
+By default we use step learning rate with 1x schedule, this calls [MultiStepLR](https://github.com/open-mmlab/mmengine/blob/main/mmengine/optim/scheduler/lr_scheduler.py#L139) in MMEngine.
+We support many other learning rate schedule [here](https://github.com/open-mmlab/mmengine/blob/main/mmengine/optim/scheduler/lr_scheduler.py), such as `CosineAnnealingLR` and `PolyLR` schedule. Here are some examples
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:196]: schedule -> schedules

File: docs/en/advanced_guides/customize_runtime.md [hunk 7]
```diff
@@ -20,240 +45,268 @@ To modify the learning rate of the model, the users only need to modify the `lr`
...
+
+    def after_train_iter(self,
+                         runner,
+                         batch_idx: int,
+                         data_batch: DATA_BATCH = None,
+                         outputs: Optional[dict] = None) -> None:
 ```
 
-Depending on the functionality of the hook, the users need to specify what the hook will do at each stage of the training in `before_run`, `after_run`, `before_epoch`, `after_epoch`, `before_iter`, and `after_iter`.
+Depending on the functionality of the hook, the users need to specify what the hook will do at each stage of the training in `before_run`, `after_run`, `before_train`, `after_train` , `before_train_epoch`, `after_train_epoch`, `before_train_iter`, and `after_train_iter`.  There are more points where hooks can be inserted, refer to [base hook class](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/hook.py#L9) for more detail.
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:292]: the users -> users

File: docs/en/advanced_guides/customize_runtime.md [hunk 8]
```diff
@@ -274,60 +327,39 @@ custom_hooks = [
...
+        max_keep_ckpts=3,
+        save_optimizer=True))
 ```
 
-#### Evaluation config
+#### LoggerHook
 
-The config of `evaluation` will be used to initialize the [`EvalHook`](https://github.com/open-mmlab/mmdetection/blob/v2.13.0/mmdet/core/evaluation/eval_hooks.py#L9).
-Except the key `interval`, other arguments such as `metric` will be passed to the `dataset.evaluate()`.
+The `LoggerHook` enables to set intervals. And the detail usages can be found in the [docstring](https://github.com/open-mmlab/mmengine/blob/main/mmengine/hooks/logger_hook.py#L18).
```
Comment (Tai-Wang) [inline docs/en/advanced_guides/customize_runtime.md:361]: enables to set -> enables setting

Comment (Tai-Wang) [reply docs/en/advanced_guides/customize_runtime.md:361]: And the detail usages -> The detailed instructions

**Review by Tai-Wang — APPROVED** [review]
